### DISCUSSION TITLE
Using underwater structures as "centers"

### DISCUSSION THREAD

**Discussion by echelleburns** [bmcclintock/momentuHMM#89] [Q&A] [answered]
For MIfitHMM, there is a way to specify activity centers that may act as attractive/repulsive agents for the movement model. Is there a way to specify a more complex underwater structure, such as a pipe or large artificial reef that may be used in a similar way? Transforming a shapeline to a series of points significantly increases processing time for model fitting, as the model is then required to fit angle and distance covariates to each individual point, not just the one closest to the animal at that point in space and time. However, to ignore such a structure would bias the model output.

**Comment by bmcclintock** [comment, marked as answer]
Check out the 'centroids' argument, which allows for dynamic activity centers where the coordinates can change over time.  In your case, you would need to manually identify the centroid for each time step (i.e., the nearest point on the structure to the animal).  This could be done by: 1) running MIfitHMM with fit=FALSE to get the imputed tracks; 2) identifying the corresponding location for each time step for each imputed track in the returned crwSim$miData list object; 3) re-running prepData on each imputed track with the 'centroids' argument indicating the nearest point along the structure for each time step; 4) running MIfitHMM with fit=TRUE on the modified list of imputed momentuHMMData objects.  

Alternatively, you could probably get fancy and write a function that identifies the nearest point along the structure and calculates the corresponding distance and/or angle.  You could then simply use this function in the formula and/or DM arguments for MIfitHMM.
